src/struct.c

```diff
@@ -631,6 +631,52 @@ Dsymbol *AggregateDeclaration::searchCtor()
     return s;
 }
 
+/***************************************
+ * Return true if aggregate is an InputRange.
+ */
+bool AggregateDeclaration::isInputRange()
+{
+    return (this->search(Loc(), Id::Ffront) &&
```

Comment:
Unless we replace them *all* with `isXType` or `dynamic_cast` that is...I'm simply wishing the DMD code base to one day become as clean in readability as Clang is.